titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/iam/AwsAssumePrincipal.java

```diff
@@ -39,6 +40,9 @@ public AwsAssumePrincipal(@JsonProperty("Principal") List<String> principals) {
     }
 
     public List<String> getPrincipals() {
+        if (null == principals) {
```

Comment:
As this object is immutable, this check can be done in the constructor.

Reply:
Good point.